internal/services/network/route_server_resource.go

```diff
@@ -132,6 +143,7 @@ func resourceRouteServerCreate(d *pluginsdk.ResourceData, meta interface{}) erro
 		Properties: &virtualwans.VirtualHubProperties{
 			Sku:                        pointer.To(d.Get("sku").(string)),
 			AllowBranchToBranchTraffic: pointer.To(d.Get("branch_to_branch_traffic_enabled").(bool)),
+			HubRoutingPreference:       (*virtualwans.HubRoutingPreference)(pointer.To(d.Get("hub_routing_preference").(string))),
```

Comment:
Can we change this to 
```suggestion
			HubRoutingPreference:       pointer.To(virtualwans.HubRoutingPreference(d.Get("hub_routing_preference").(string))),
```